src/orders/components/OrderCustomerAddressesEditDialog/OrderCustomerAddressesEditDialog.tsx

```diff
@@ -0,0 +1,368 @@
+import { useStyles } from "./styles";
+
+export interface OrderCustomerAddressesEditDialogOutput {
+  shippingAddress: AddressInput;
+  billingAddress: AddressInput;
+}
+
+export interface OrderCustomerAddressesEditDialogProps {
+  open: boolean;
+  confirmButtonState: ConfirmButtonTransitionState;
+  errors: OrderErrorFragment[];
+  countries?: Array<{
+    code: string;
+    label: string;
+  }>;
+  customerAddresses?: CustomerAddresses_user_addresses[];
+  defaultShippingAddress?: CustomerAddresses_user_defaultShippingAddress;
+  defaultBillingAddress?: CustomerAddresses_user_defaultBillingAddress;
+  onClose();
+  onConfirm(data: OrderCustomerAddressesEditDialogOutput): SubmitPromise;
+}
+
+const OrderCustomerAddressesEditDialog: React.FC<OrderCustomerAddressesEditDialogProps> = props => {
+  const {
+    open,
+    confirmButtonState,
+    errors = [],
+    countries = [],
+    customerAddresses = [],
+    defaultShippingAddress,
+    defaultBillingAddress,
+    onClose,
+    onConfirm
+  } = props;
+
+  const classes = useStyles(props);
+  const intl = useIntl();
+  const {
+    errors: shippingValidationErrors,
+    submit: handleShippingSubmit
+  } = useAddressValidation(address => address, AddressTypeEnum.SHIPPING);
+  const {
+    errors: billingValidationErrors,
+    submit: handleBillingSubmit
+  } = useAddressValidation(address => address, AddressTypeEnum.BILLING);
+  const dialogErrors = useModalDialogErrors(
+    [...errors, ...shippingValidationErrors, ...billingValidationErrors],
+    open
+  );
+
+  const getCustomerAddress = (customerAddressId: string): AddressInput =>
+    transformAddressToAddressInput(
+      customerAddresses.find(
+        customerAddress => customerAddress.id === customerAddressId
+      )
+    );
+
+  const handleAddressesSubmit = (data: OrderCustomerAddressesEditFormData) => {
+    const shippingAddress =
+      data.shippingAddressInputOption ===
+      AddressInputOptionEnum.CUSTOMER_ADDRESS
+        ? getCustomerAddress(data.customerShippingAddress.id)
+        : handleShippingSubmit(data.shippingAddress);
+
+    if (data.billingSameAsShipping) {
+      return {
+        shippingAddress,
+        billingAddress: shippingAddress
+      };
+    }
+
+    const billingAddress =
+      data.billingAddressInputOption === AddressInputOptionEnum.CUSTOMER_ADDRESS
+        ? getCustomerAddress(data.customerBillingAddress.id)
+        : handleBillingSubmit(data.billingAddress);
+
+    return {
+      shippingAddress,
+      billingAddress
+    };
+  };
+
+  const handleSubmit = (data: OrderCustomerAddressesEditFormData) => {
+    const adressesInput = handleAddressesSubmit(data);
+
+    if (adressesInput.shippingAddress && adressesInput.billingAddress) {
+      onConfirm(adressesInput);
+    }
+  };
+
+  const countryChoices = countries.map(country => ({
+    label: country.label,
+    value: country.code
+  }));
+
+  return (
+    <Dialog onClose={onClose} open={open}>
+      <OrderCustomerAddressesEditForm
```

Comment:
Can we make some smaller components of out this?

Reply:
Somehow changed